exporter/trace/stackdriver/trace.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2019, OpenTelemetry Authors
+// Copyright 2017, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackdriver
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	traceclient "cloud.google.com/go/trace/apiv2"
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/api/support/bundler"
+	tracepb "google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
+
+	"go.opentelemetry.io/sdk/trace"
+)
+
+// traceExporter is an imeplementation of trace.Exporter
+// that uploads spans to Stackdriver Trace.
+type traceExporter struct {
+	o         *options
+	projectID string
+	bundler   *bundler.Bundler
+	// uploadFn defaults in uploadSpans; it can be replaced for tests.
+	uploadFn func(spans []*tracepb.Span)
+	overflowLogger
+	client *traceclient.Client
+}
+
+func newTraceExporter(o *options) (*traceExporter, error) {
+	ctx := o.Context
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	client, err := traceclient.NewClient(ctx, o.TraceClientOptions...)
+	if err != nil {
+		return nil, fmt.Errorf("Stackdriver: couldn't initiate trace client: %v", err)
+	}
+	return newTraceExporterWithClient(o, client), nil
+}
+
+const defaultBufferedByteLimit = 0 * 1024 * 1024
+
+func newTraceExporterWithClient(o *options, c *traceclient.Client) *traceExporter {
+	e := &traceExporter{
+		projectID: o.ProjectID,
+		client:    c,
+		o:         o,
+	}
+	b := bundler.NewBundler((*tracepb.Span)(nil), func(bundle interface{}) {
+		e.uploadFn(bundle.([]*tracepb.Span))
+	})
```

Comment:
bundler can be removed with BatchSpanProcessor.

Reply:
In that case, my understanding is that ExportSpans need to implement all implementations to upload SpanData in batch safely. Is that correct? It seems Jaeger exporter is still using bundler for SimpleSpanProcessor.
https://github.com/open-telemetry/opentelemetry-go/blob/master/exporter/trace/jaeger/jaeger.go#L104-L108

Reply:
Jaeger should be updated to use BatchSpanProcessor as well.

Reply:
google.golang.org/api/support/bundler was totally removed.